File: app/src/main/java/com/example/miniapp/viewmodels/NewTaskViewModel.java
package com.example.miniapp.viewmodels;

import android.annotation.SuppressLint;
import android.net.Uri;

import com.example.miniapp.helper_classes.Logger;
import com.example.miniapp.models.IUserDBManager;

import java.text.DateFormatSymbols;
import java.util.Calendar;
import java.util.Date;

public class NewTaskViewModel {
    private IUserDBManager dbManager;

    public NewTaskViewModel(IUserDBManager dbManager){
        this.dbManager = dbManager;
    }

    public static String  dateRepresentation(int year, int month, int day) {
        // needs localization
        String monthName = new DateFormatSymbols().getMonths()[month];

        return day + " " + monthName + " " + year;
    }

    @SuppressLint("DefaultLocale")
    public static String timeRepresentation(int hr, int min) {
        int hour = hr % 12;
        if (hour == 0) hour = 12;

        return String.format("%02d:%02d %s", hour, min, hr < 12 ? "am" : "pm");

    }

    public void openDB(){
        dbManager.openDB();
    }

    public void closeDB(){
        dbManager.closeDB();
    }

    public void submit(String task, Date created, Date start, Uri imageURI, String completeAddress){

        String imageURIString = imageURI == null ? "" : imageURI.toString();
        String address = completeAddress == null ? "" : completeAddress;

        dbManager.create(task, created, start, imageURIString, address);

        Logger.log("Task: " + task);
        Logger.log("Created: " + created);
        Logger.log("Start: " + start);
        Logger.log("URI: " + imageURIString);
        Logger.log("Address: " + address);
    }

    public boolean isValid(Date dateTimeSelected) {
        Date now = Calendar.getInstance().getTime();
        return dateTimeSelected.after(now);
    }
}

File: app/src/main/java/com/example/miniapp/viewmodels/NewTaskViewModel.kt
package com.example.miniapp.viewmodels

import android.annotation.SuppressLint
import android.net.Uri
import com.couchbase.lite.MutableDocument
import com.example.miniapp.models.IDBManager
import java.text.DateFormatSymbols
import java.util.*

class NewTaskViewModel(private val dbManager: IDBManager) {
    fun submit(task: String, dateCreated: Date, dateStart: Date, imageURI: Uri?, completeAddress: String?) {
        val imageURIString = imageURI?.toString() ?: ""
        val address = completeAddress ?: ""

        dbManager.create {
            val doc = MutableDocument()
            doc.setString("task", task)
            doc.setDate("dateCreated", dateCreated)
            doc.setDate("dateStart", dateStart)
            doc.setBoolean("isDone", false)
            doc.setString("imageURI", imageURIString)
            doc.setString("address", address)
        }
    }

    fun isValid(dateTimeSelected: Date): Boolean {
        val now = Calendar.getInstance().time
        return dateTimeSelected.after(now)
    }

    companion object {
        @JvmStatic
        fun dateRepresentation(year: Int, month: Int, day: Int): String {
            // needs localization
            val monthName = DateFormatSymbols().months[month]
            return "$day $monthName $year"
        }

        @JvmStatic
        @SuppressLint("DefaultLocale")
        fun timeRepresentation(hr: Int, min: Int): String {
            var hour = hr % 12
            if (hour == 0) hour = 12
            return String.format("%02d:%02d %s", hour, min, if (hr < 12) "am" else "pm")
        }
    }
}